plot_util_test.py

```diff
@@ -1,77 +1,58 @@
-
 from plot_util import GB
 
 import os
 import pyfakefs
-import unittest
 
 import plot_util
 
-class TestPlotUtil(unittest.TestCase):
-    def test_human_format(self):
-        self.assertEqual('3G', plot_util.human_format(3442000000, 0))
-        self.assertEqual('3.54M', plot_util.human_format(3542000, 2))
-        self.assertEqual('354', plot_util.human_format(354, 0))
-        self.assertEqual('354.00', plot_util.human_format(354, 2))
-
-    def test_time_format(self):
-        self.assertEqual('34s', plot_util.time_format(34))
-        self.assertEqual('59s', plot_util.time_format(59))
-        self.assertEqual('0:01', plot_util.time_format(60))
-        self.assertEqual('0:01', plot_util.time_format(119))
-        self.assertEqual('0:02', plot_util.time_format(120))
-        self.assertEqual('1:01', plot_util.time_format(3694))
-
-    def test_split_path_prefix(self):
-        self.assertEqual(
-                ('', []),
-                plot_util.split_path_prefix( [] ) )
-        self.assertEqual(
-                ('', ['/a/0', '/b/1', '/c/2']),
-                plot_util.split_path_prefix([ '/a/0', '/b/1', '/c/2' ]) )
-        self.assertEqual(
-                ('/a/b', ['0', '1', '2']),
-                plot_util.split_path_prefix([ '/a/b/0', '/a/b/1', '/a/b/2' ]) )
-
-    def test_columns(self):
-        self.assertEqual(
-                [ [ 0, 3, 6 ],
-                  [ 1, 4, 7 ],
-                  [ 2, 5, '--'] ],
-                plot_util.column_wrap(list(range(8)), 3, filler='--') )
-        self.assertEqual(
-                [ [ 0, 3, 6 ],
-                  [ 1, 4, 7 ],
-                  [ 2, 5, 8 ] ],
-                plot_util.column_wrap(list(range(9)), 3, filler='--') )
-        self.assertEqual(
-                [ [ 0 ],
-                  [ 1 ],
-                  [ 2 ] ],
-                plot_util.column_wrap(list(range(3)), 1, filler='--') )
-
-
-class TestPlotUtilFiles(TestCase):
-    def setUp(self):
-        self.setUpPyfakefs()
-
-    def test_list_k32_plots(self):
-        self.fs.create_file('/t/plot-k32-0.plot', st_size=108 * GB)
-        self.fs.create_file('/t/plot-k32-1.plot', st_size=108 * GB)
-        self.fs.create_file('/t/.plot-k32-2.plot', st_size=108 * GB)
-        self.fs.create_file('/t/plot-k32-3.plot.2.tmp', st_size=108 * GB)
-        self.fs.create_file('/t/plot-k32-4.plot', st_size=100 * GB)
-        self.fs.create_file('/t/plot-k32-5.plot', st_size=108 * GB)
-
-        self.assertEqual( [ '/t/plot-k32-0.plot',
-                            '/t/plot-k32-1.plot',
-                            '/t/plot-k32-5.plot' ],
-                plot_util.list_k32_plots('/t/') )
-
-
-if __name__ == '__main__':
-    unittest.main()
+def test_human_format():
+    assert (plot_util.human_format(3442000000, 0) == '3G')
+    assert (plot_util.human_format(3542000, 2) == '3.54M')
+    assert (plot_util.human_format(354, 0) == '354')
+    assert (plot_util.human_format(354, 2) == '354.00')
```

Comment:
I dunno...  so the repetition isn't particularly an issue here but if the first fails you get no other feedback from which to derive a pattern of failure.

```suggestion
@pytest.mark.parametrize(
	argnames=['inhuman', 'human'],
	argvalues=[
	    [[3442000000, 0], '3G'],
	    [[3542000, 2], '3.54M'],
	    [[354, 0], '354'],
	    [[354, 2], '354.00'],
    ],
)
def test_human_format(inhuman, human):
	assert plot_util.human_format(*inhuman) == human
```

Also, is [humanize](https://python-humanize.readthedocs.io/en/latest/filesize/) what you want?  I didn't dig into MiB vs. whatever, maybe it isn't.  Also related, [pint](https://pint.readthedocs.io/en/stable/) for more general units handling.

Reply:
regarding the libraries, they might be what i want.  let's figure out how to have a sane install process before i go adding more dependencies, though.

re: the other, yeah I didn't think about that.  I'm used to "expects" where you can list a bunch of them in a row.  I forgot switching to assert means it will bail on the first failure.  Is there a way to avoid that with pytest?  What you suggest could work, but i have a preference for very simple test code.

Reply:
https://pypi.org/project/expects/?

Reply:
Hmm...  I can certainly agree that my suggestion is not beautiful.  I don't offhand know an answer but did search for 'delayed assert' and lo and behold that was actually a relevant set of terms.  Found a few other related things.  Haven't dug into which would be 'best' or how great they really are.

https://pypi.org/project/delayed-assert/
https://pypi.org/project/pytest-assume/
https://github.com/pytest-dev/pytest/issues/1367

Reply:
Hmm, ok.  Something to look into.